singularity-eos/eos/modifiers/ramps_eos.hpp

```diff
@@ -0,0 +1,163 @@
+    return t_.GruneisenParamFromDensityInternalEnergy(rho, sie, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real PressureFromDensityTemperature(const Real rho, const Real temperature,
+                                      Real *lambda = nullptr) const {
+    const Real p_ramp {get_ramp_pressure(rho)};
+    const Real p_eos {t_.PressureFromDensityTemperature(rho, temperature, lambda)};
+    return p_eos < p_ramp ? p_ramp : p_eos;
+  }
+  PORTABLE_FUNCTION
+  Real SpecificHeatFromDensityTemperature(const Real rho, const Real temperature,
+                                          Real *lambda = nullptr) const {
+    return t_.SpecificHeatFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real BulkModulusFromDensityTemperature(const Real rho, const Real temperature,
+                                         Real *lambda = nullptr) const {
+    return t_.BulkModulusFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  Real GruneisenParamFromDensityTemperature(const Real rho, const Real temperature,
+                                            Real *lambda = nullptr) const {
+    return t_.GruneisenParamFromDensityTemperature(rho, temperature, lambda);
+  }
+  PORTABLE_FUNCTION
+  void FillEos(Real &rho, Real &temp, Real &energy, Real &press, Real &cv, Real &bmod,
+               const unsigned long output, Real *lambda = nullptr) const {
+    Real senergy;
+    switch (t_.PreferredInput()) {
+    case thermalqs::density | thermalqs::temperature:
+      t_.FillEos(rho, temp, energy, press, cv, bmod, output, lambda);
+      energy = energy + shift_;
+      break;
+    case thermalqs::density | thermalqs::specific_internal_energy:
+      senergy = energy - shift_;
+      t_.FillEos(rho, temp, senergy, press, cv, bmod, output, lambda);
+      break;
+    default:
+      EOS_ERROR("Didn't find a valid input for SAPRampEOS::FillEOS\n");
+    }
+  }
+
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return t_.nlambda(); }
+
+  PORTABLE_FUNCTION
+  unsigned long PreferredInput() const { return t_.PreferredInput(); }
+
+  PORTABLE_FUNCTION void PrintParams() const {
+    t_.PrintParams();
+    printf("scaling_ratio = %f\n", shift_);
+  }
+  PORTABLE_FUNCTION
+  void DensityEnergyFromPressureTemperature(const Real press, const Real temp,
+                                            Real *lambda, Real &rho, Real &sie) const {
+    t_.DensityEnergyFromPressureTemperature(press, temp, lambda, rho, sie);
+    sie = sie + shift_;
+  }
+
+  PORTABLE_FUNCTION
+  void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                              Real &bmod, Real &dpde, Real &dvdt,
+                              Real *lambda = nullptr) const {
+    t_.ValuesAtReferenceState(rho, temp, sie, press, cv, bmod, dpde, dvdt, lambda);
+    sie += shift_;
+  }
+
+  // Vector functions that overload the scalar versions declared here.
+  SG_ADD_BASE_CLASS_USINGS(SAPRampEOS<T>)
```

Comment:
specific overloads might be necessary

Reply:
Won't the vector calls just access the scalar versions where the modifier is already applied?